### DISCUSSION TITLE
Apatch on an already modified boot.img (Samsung)

### DISCUSSION THREAD

**Discussion by dazzleworth** [bmax121/APatch#621] [Q&A]
I have a Samsung device with an already unlocked bootloader and used chimeratool to perform a Knoxguard removed. I can tell that chimeratool modified the boot img with an red warning text upon boot that phone is not running official firmware (after bootloader unlocked) warning. No bootloop and phone boots up fine. No root access also. I'm able to obtain the current modified boot image state using DSU loader and plan to use apatch to patch that. Can I restore modified boot.img before apatch  (on Odin or TWRP) if I encounter bootloop? I don't want to restore stock firmware. Thanks.

System info:

![Screenshot_20240821-184402_Settings](https://github.com/user-attachments/assets/751bca9d-2523-433d-b49f-41f46ff96f78)

**Comment by Many2050** [comment]
Greetings, in my case I restored the modified boot.img several times without any problem. Were you able to apply apatch without problems? Samsungs are awesome. In my case, apply apatch and I get boot loop?

**Reply by dazzleworth** [reply]
Yes! I had to flash a custom kernel, in this case I picked HariKernel (the only one compatible with Exynos 2100  devices) for OneUi  (stock rom) then I could patch boot.img with Apatch without bootloop
